### PR TITLE
Rdm-4535 Case List Filters new contract

### PR DESCRIPTION
**Pull request by noronhaa**
### Change description ###
Move Case List filters to toolkit and implements new contract 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/app/workbasket/filters/workbasket-filters.component.spec.ts [hunk 1]
```diff
@@ -1,1058 +0,0 @@
...
-  let router: any;
-  let activatedRoute: any;
-  let jurisdictionService: JurisdictionService;
-  let workbasketInputFilterService: any;
-  let orderService: any;
-  let alertService: AlertService;
-  let windowService;
-  const TEST_FORM_GROUP = new FormGroup({});
-
-  describe('Clear localStorage for workbasket filters', () => {
```
Comment (andrewfolga) [inline src/app/workbasket/filters/workbasket-filters.component.spec.ts]: These tests are removed here and not added in the ui toolkit for case-list-filters.components.ts. Are we getting rid of them?

File: src/app/workbasket/filters/workbasket-filters.component.spec.ts [hunk 2]
```diff
@@ -1,1058 +0,0 @@
...
-      component.selected.jurisdiction = JURISDICTION_2;
-      component.selected.caseType = null;
-      component.selected.caseState = null;
-
-      fixture.detectChanges();
-      let button = de.query($APPLY_BUTTON);
-      expect(button.nativeElement.disabled).toBeTruthy();
-    }));
-
-    it('should have an Apply button disabled when case type is not set', async(() => {
```
Comment (andrewfolga) [inline src/app/workbasket/filters/workbasket-filters.component.spec.ts]: Some other tests like this one assessing the buttons being disabled are missing in `case-list-filters.component.spec.ts`. Are we sure we do not want to keep them?

File: src/app/workbasket/workbasket.component.ts [hunk 3]
```diff
@@ -94,6 +98,14 @@ export class WorkbasketComponent implements OnInit {
         });
 
     this.scrollToTop();
+
+    this.router.navigate(['/list/case'], {
```
Comment (andrewfolga) [inline src/app/workbasket/workbasket.component.ts]: Do we have a test case asserting this action?

File: src/app/workbasket/workbasket.component.ts [hunk 4]
```diff
@@ -94,6 +98,14 @@ export class WorkbasketComponent implements OnInit {
...
+      queryParams: returnValue.queryParams
+    });
+  }
+
+  applyReset() {
+    this.router.navigate(['/list/case']);
```
Comment (andrewfolga) [inline src/app/workbasket/workbasket.component.ts]: Can we have a test case asserting this action?

File: src/app/workbasket/workbasket.component.ts (hunk 3 above)
Comment (adnanakgun) [reply src/app/workbasket/workbasket.component.ts]: Added now

File: src/app/workbasket/workbasket.component.ts (hunk 4 above)
Comment (adnanakgun) [reply src/app/workbasket/workbasket.component.ts]: Added now

**Pull request merged**